### PR TITLE
[bitnami/grafana-tempo] Add metrics-generator service

### PR DESCRIPTION
**Pull request by symmetriac**
Signed-off-by: Artem Belov <artem.belov@pm.bet>

### Description of the change

New service metrics-generator was added

### Benefits

Metrics-generator deployment can collect metrics from your spans

### Applicable issues

  - fixes #

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/grafana-tempo/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ name: grafana-tempo
 sources:
   - https://github.com/bitnami/bitnami-docker-grafana-tempo
   - https://github.com/grafana/tempo/
-version: 1.1.6
+version: 1.1.7
```
Comment (carrodher) [inline bitnami/grafana-tempo/Chart.yaml:31]: Since it is a feature addition instead of bug fix, I would bump the minor version instead of the patch one

**Review by carrodher — COMMENTED** [review]
Thanks for this PR, could you please check other charts already using metrics such as the regular grafana in order to replicate the logic/structure followed in the rest of the charts?

File: bitnami/grafana-tempo/Chart.yaml (hunk 1 above)
Comment (symmetriac) [reply bitnami/grafana-tempo/Chart.yaml:31]: > Thanks for this PR, could you please check other charts already using metrics such as the regular grafana in order to replicate the logic/structure followed in the rest of the charts?

Unfortunately I couldn't find any samples of how to deploy this new component, except this one in grafana repo
https://github.com/grafana/tempo/blob/main/operations/kube-manifests/Deployment-metrics-generator.yaml

Comment (carrodher) [reply bitnami/grafana-tempo/Chart.yaml:31]: I mean how the metrics-related k8s objects (in the `/templates` folder) are being defined. In the same way, the `metrics:` parameters in the `values.yaml` follow a similar structure in all the Bitnami Helm charts.

Comment (symmetriac) [reply bitnami/grafana-tempo/Chart.yaml]: Sorry, I don't follow what you are trying to say :) 
I've added the templates for new tempo service, `metrics:` parameter in the values.yaml is defined for deploy servicemonitors for services. Maybe you can give me an example?

Comment (carrodher) [reply bitnami/grafana-tempo/Chart.yaml]: I mean following the same approach used in other Bitnami Helm charts, for example in the Grafana one:
- `values.yaml`: https://github.com/bitnami/charts/blob/7d41614a6e43ff6118a4310a3e72cf230bce94a8/bitnami/grafana/values.yaml#L714
- README: https://github.com/bitnami/charts/blob/master/bitnami/grafana/README.md#metrics-parameters
- Everything related with metrics at templates:
  - https://github.com/bitnami/charts/blob/master/bitnami/grafana/templates/image-renderer-prometheusrules.yaml
  - https://github.com/bitnami/charts/blob/master/bitnami/grafana/templates/image-renderer-servicemonitor.yaml
  - https://github.com/bitnami/charts/blob/master/bitnami/grafana/templates/image-renderer-deployment.yaml
  - https://github.com/bitnami/charts/blob/master/bitnami/grafana/templates/prometheusrules.yaml
  - https://github.com/bitnami/charts/blob/master/bitnami/grafana/templates/servicemonitor.yaml

Comment (symmetriac) [reply bitnami/grafana-tempo/Chart.yaml]: But we do have a metrics parameter in tempo values:
https://github.com/bitnami/charts/blob/master/bitnami/grafana-tempo/values.yaml#L2111
https://github.com/bitnami/charts/tree/master/bitnami/grafana-tempo#metrics-parameters
And the related servicemonitors deploy based on these values
https://github.com/bitnami/charts/blob/master/bitnami/grafana-tempo/templates/compactor/servicemonitor.yaml#L1
https://github.com/bitnami/charts/blob/master/bitnami/grafana-tempo/templates/distributor/servicemonitor.yaml#L1
If you think that we should do it like in Image Renderer in Grafana chart, `.Values.imageRenderer.metrics.enabled` and `.Values.imageRenderer.metrics.serviceMonitor.enabled` than it should be done in every tempo services.

File: bitnami/grafana-tempo/values.yaml [hunk 2]
```diff
@@ -771,6 +781,268 @@ distributor:
...
+    ## ref https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip
+    ##
+    externalTrafficPolicy: Cluster
+    ## @param metricsGenerator.service.annotations Additional custom annotations for metricsGenerator service
+    ##
+    annotations: {}
+    ## @param metricsGenerator.service.extraPorts Extra ports to expose in the metricsGenerator service
+    ##
+    extraPorts: []
+
```
Comment (carrodher) [inline bitnami/grafana-tempo/values.yaml:1044]:
```suggestion
    extraPorts: []
```

File: bitnami/grafana-tempo/values.yaml [hunk 3]
```diff
@@ -771,6 +781,268 @@ distributor:
     ##
     extraPorts: []
 
+## @section metricsGenerator Deployment Parameters
```
Comment (carrodher) [inline bitnami/grafana-tempo/values.yaml:784]:
```suggestion
## @section Metrics Generator Deployment Parameters
```

File: bitnami/grafana-tempo/values.yaml [hunk 4]
```diff
@@ -771,6 +781,268 @@ distributor:
...
+  ## e.g:
+  ## initContainers:
+  ##  - name: your-image-name
+  ##    image: your-image
+  ##    imagePullPolicy: Always
+  ##    command: ['sh', '-c', 'echo "hello world"']
+  ##
+  initContainers: []
+
+  ## @section metricsGenerator Traffic Exposure Parameters
```
Comment (carrodher) [inline bitnami/grafana-tempo/values.yaml:991]:
```suggestion
  ## @section Metrics Generator Traffic Exposure Parameters
```

**Review by carrodher — COMMENTED** [review]
I see your point after taking a look carefully. I agree with the changes, just a couple of minor comments before merging the PR

**Review by carrodher — APPROVED** [review]
LGTM

**Comment by symmetriac** [comment]
@carrodher thanks!

**Pull request merged**